File: manicode_wrapper.ts
import * as pty from "node-pty";
import * as os from "node:os";
import path from "node:path";

async function executeManicode(
  instructions: string,
  options: {
    cwd: string;
  }
 ): Promise<string> {
  return new Promise((resolve, reject) => {
    let output = "";

    const shell = os.platform() === "win32" ? "powershell.exe" : "bash";

    const ptyProcess = pty.spawn(shell, [], {
      name: "xterm-color",
      cols: 80,
      rows: 30,
      cwd: options.cwd,
      env: process.env,
    });

    ptyProcess.onData((data) => {
      output += data;

      if (output.includes("Complete!")) {
        ptyProcess.kill();
      }
    });

    ptyProcess.write(`manicode . "${instructions}"\r`);

    ptyProcess.onExit(() => {
      resolve(output);
    });
  });
}

async function test() {
  const projectRoot = path.join(process.cwd(), "../template");

  const result = await executeManicode(
    "double the font size of all text on the homepage",
    {
      cwd: projectRoot,
    }
  );

  console.log(result);
}

test();

File: manicode_wrapper.py
"""Test wrapper for executing manicode commands"""
import os
import platform
from typing import Dict
import asyncio
import ptyprocess

async def execute_manicode(instructions: str, options: Dict[str, str]) -> str:
    """Execute a test manicode command"""
    output = ""

    shell = "powershell.exe" if platform.system() == "Windows" else "bash"

    pty = ptyprocess.PtyProcess.spawn(
        [shell],
        env=os.environ,
        cwd=options["cwd"],
        dimensions=(30, 80)
    )

    pty.write(f'manicode . "{instructions}"\r')

    while pty.isalive():
        try:
            data = pty.read()
            if data:
                output += data.decode()
                if "Complete!" in output:
                    pty.terminate()
                    break
        except EOFError:
            break

    return output

async def test():
    """Test function"""
    project_root = os.path.join(os.getcwd(), "../template")

    result = await execute_manicode(
        "double the font size of all text on the homepage",
        {"cwd": project_root}
    )

    print(result)

if __name__ == "__main__":
    asyncio.run(test())